rule.proto

```diff
@@ -0,0 +1,221 @@
+syntax = "proto2";
```

Comment:
why not use proto3

Reply:
The proto Jin created uses proto2, so I thought I'd keep it consistent. I don't really use any features that proto2 has that proto3 doesn't, so should I change it?